optuna_integration/sklearn/sklearn.py

```diff
@@ -303,10 +303,12 @@ def _cross_validate_with_pruning(
 
         for step in range(self.max_iter):
             for i, (train, test) in enumerate(self.cv.split(self.X, self.y, groups=self.groups)):
-                out = self._partial_fit_and_score(estimators[i], train, test, partial_fit_params)
+                _out = self._partial_fit_and_score(estimators[i], train, test, partial_fit_params)
+                out = np.asarray(_out, dtype=np.float64)
```

Comment:
If this change is allowed, I think it would be much easier to replace `Number` with `float` in this file.